File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        Queue<Integer> q = new LinkedList<>();
        for (int i = 1; i <= N; i++) {
            q.add(i);
        }

        StringBuilder sb = new StringBuilder();
        sb.append("<");

        while (!q.isEmpty()) {
            for (int i = 0; i < K - 1; i++) {
                q.add(q.poll()); // 앞에서 빼서 뒤에 넣기
            }
            sb.append(q.poll()); // K번째 제거
            if (!q.isEmpty()) {
                sb.append(", ");
            }
        }

        sb.append(">");
        System.out.println(sb.toString());
    }
}

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.kt
import java.util.LinkedList

fun main() {
    val (n, k) = readln().split(" ").map { it.toInt() }
    val q: LinkedList<Int> = LinkedList((1..n).toList())

    val result = mutableListOf<Int>()

    while (q.isNotEmpty()) {
        repeat(k - 1) {
            q.add(q.poll()) // 앞에서 빼서 뒤에 넣기
        }
        result.add(q.poll()) // K번째 제거
    }

    println("<${result.joinToString(", ")}>")
}